WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -236,6 +236,36 @@ def process_payment
     redirect_to competition_register_path
   end
 
+  private def journaled_stripe_charge(*stripe_charge_create_args)
+    # Talk to Stripe to make a charge, but "journal" the attempt.
+    #  1. Before talking to Stripe, we first create a StripeCharge in status "unknown".
+    #  2. Talk to Stripe.
+    #  3. After talking to Stripe, we update that StripeCharge to status to "success" or "failure".
+    # If anything else happens (maybe our server crashed during step 2, or
+    # between step 2 and 3), then we'll have a StripeCharge in our database
+    # with "unknown" status, and we'll know to investigate it, most likely by
+    # visiting our Stripe dashboard to see if the charge actually happened or not.
+
+    stripe_charge = StripeCharge.create!(
+      metadata: stripe_charge_create_args.to_json,
+      stripe_charge_id: nil,
+      status: "unknown",
+    )
+
+    begin
+      charge = Stripe::Charge.create(*stripe_charge_create_args)
+      stripe_charge.update!(
+        status: "success",
+        stripe_charge_id: charge.id,
+      )
+    rescue Stripe::CardError => e
```

Comment:
Can we log this error somehow?
I remember a couple of weeks ago we got an issue because a connected Stripe account was not fully activated: the user got the "famous" 500 error page, and we had to get deep in the log to find the root cause.
This would allow for an easy debug until #3448 is implemented.

Reply:
Oh right, but I'd actually be up storing any error occurring with the charge and raising it again: the caller could still act upon it, but at least we would have stored anything that went wrong.

Maybe simply adding an extra field to the table, and store the error there when it occurs would be enough?

Reply:
Oh right, but I'd actually be up storing any error occurring with the charge and raising it again: the caller could still act upon it, but at least we would have stored anything that went wrong.

Maybe simply adding an extra field to the table, and store the error there when it occurs would be enough?

Reply:
Adding a new column with some details about the relevant error sounds useful! I've gone ahead and made that change. Please take another look and let me know what you think!